<a href="https://colab.research.google.com/github/MarcoBS14/Seminario-de-Ciencia-de-datos/blob/main/SeminarioCD3_intro_pandas_p3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [82]:
import pandas as pd

In [83]:
salarios=pd.read_csv("/Users/marcobarragan/Documents/Seminario-de-Ciencia-de-datos/salary.csv")

In [84]:
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  
0          2174             0              40   United-States   <=50K  
1             0             0              13   United-States   <=50K  
2             0             0              40   United-States   <=50K  
3             0             0              40   United-States   <=50K  
4             0             0              40            Cuba   <=50K

In [85]:
salarios.tail()

age      workclass  fnlwgt    education  education-num  \
32556   27        Private  257302   Assoc-acdm             12   
32557   40        Private  154374      HS-grad              9   
32558   58        Private  151910      HS-grad              9   
32559   22        Private  201490      HS-grad              9   
32560   52   Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation relationship    race      sex  \
32556   Married-civ-spouse        Tech-support         Wife   White   Female   
32557   Married-civ-spouse   Machine-op-inspct      Husband   White     Male   
32558              Widowed        Adm-clerical    Unmarried   White   Female   
32559        Never-married        Adm-clerical    Own-child   White     Male   
32560   Married-civ-spouse     Exec-managerial         Wife   White   Female   

       capital-gain  capital-loss  hours-per-week  native-country  salary  
32556             0             0              38   United-States   <=50K  
32557             0             0              40   United-States    >50K  
32558             0             0              40   United-States   <=50K  
32559             0             0              20   United-States   <=50K  
32560         15024             0              40   United-States    >50K

In [86]:
salarios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  salary          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


In [87]:
var_cat=['salary', 'workclass', 'education','marital-status', 'occupation', 'relationship', 'race', 'sex', 'native-country'] #lista de variables que deberían tener categorías
for variable in var_cat: #para cada elemento de la lista var_cat
    salarios[variable] = salarios[variable].astype('category') #convertir la variable a categorías

In [88]:
salarios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   age             32561 non-null  int64   
 1   workclass       32561 non-null  category
 2   fnlwgt          32561 non-null  int64   
 3   education       32561 non-null  category
 4   education-num   32561 non-null  int64   
 5   marital-status  32561 non-null  category
 6   occupation      32561 non-null  category
 7   relationship    32561 non-null  category
 8   race            32561 non-null  category
 9   sex             32561 non-null  category
 10  capital-gain    32561 non-null  int64   
 11  capital-loss    32561 non-null  int64   
 12  hours-per-week  32561 non-null  int64   
 13  native-country  32561 non-null  category
 14  salary          32561 non-null  category
dtypes: category(9), int64(6)
memory usage: 1.8 MB


In [89]:
salarios.describe(include='category')

workclass education       marital-status       occupation relationship  \
count      32561     32561                32561            32561        32561   
unique         9        16                    7               15            6   
top      Private   HS-grad   Married-civ-spouse   Prof-specialty      Husband   
freq       22696     10501                14976             4140        13193   

          race    sex  native-country  salary  
count    32561  32561           32561   32561  
unique       5      2              42       2  
top      White   Male   United-States   <=50K  
freq     27816  21790           29170   24720

**Workclass** - esta variable nos indica el tipo de empresa, hay 9 tipos diferentes y la frecuencia más alta es el tipo privado con un total de 22696 registros



In [90]:
salarios['relationship']

0         Not-in-family
1               Husband
2         Not-in-family
3               Husband
4                  Wife
              ...      
32556              Wife
32557           Husband
32558         Unmarried
32559         Own-child
32560              Wife
Name: relationship, Length: 32561, dtype: category
Categories (6, object): [' Husband', ' Not-in-family', ' Other-relative', ' Own-child', ' Unmarried', ' Wife']

In [91]:
salarios.describe()

age        fnlwgt  education-num  capital-gain  capital-loss  \
count  32561.000000  3.256100e+04   32561.000000  32561.000000  32561.000000   
mean      38.581647  1.897784e+05      10.080679   1077.648844     87.303830   
std       13.640433  1.055500e+05       2.572720   7385.292085    402.960219   
min       17.000000  1.228500e+04       1.000000      0.000000      0.000000   
25%       28.000000  1.178270e+05       9.000000      0.000000      0.000000   
50%       37.000000  1.783560e+05      10.000000      0.000000      0.000000   
75%       48.000000  2.370510e+05      12.000000      0.000000      0.000000   
max       90.000000  1.484705e+06      16.000000  99999.000000   4356.000000   

       hours-per-week  
count    32561.000000  
mean        40.437456  
std         12.347429  
min          1.000000  
25%         40.000000  
50%         40.000000  
75%         45.000000  
max         99.000000

In [92]:
salarios["hours-per-day"]=salarios["hours-per-week"]/5

In [93]:
salarios["hours-per-day"].head(10)

0     8.0
1     2.6
2     8.0
3     8.0
4     8.0
5     8.0
6     3.2
7     9.0
8    10.0
9     8.0
Name: hours-per-day, dtype: float64

In [94]:
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  \
0          2174             0              40   United-States   <=50K   
1             0             0              13   United-States   <=50K   
2             0             0              40   United-States   <=50K   
3             0             0              40   United-States   <=50K   
4             0             0              40            Cuba   <=50K   

   hours-per-day  
0            8.0  
1            2.6  
2            8.0  
3            8.0  
4            8.0

In [95]:
salarios["hours-per-week"].mean()

np.float64(40.437455852092995)

In [96]:
print("El número máximo de horas por semana es de: ",salarios['hours-per-week'].max())
print("Mínimo de hrs por semana:",salarios['hours-per-week'].min())
print("Varianza de hrs por semana:",salarios['hours-per-week'].var())
print("Desviación estándar de hrs por semana: ",salarios['hours-per-week'].std())
#print("Suma de hrs por semana: ",salarios['hours-per-week'].sum())
print("Promedio de hrs por semana: ",salarios['hours-per-week'].mean())

El número máximo de horas por semana es de:  99
Mínimo de hrs por semana: 1
Varianza de hrs por semana: 152.45899505045418
Desviación estándar de hrs por semana:  12.347428681731843
Promedio de hrs por semana:  40.437455852092995


In [97]:
salarios["sex"].unique()

[' Male', ' Female']
Categories (2, object): [' Female', ' Male']

In [98]:
salarios["occupation"].unique()

[' Adm-clerical', ' Exec-managerial', ' Handlers-cleaners', ' Prof-specialty', ' Other-service', ..., ' Tech-support', ' ?', ' Protective-serv', ' Armed-Forces', ' Priv-house-serv']
Length: 15
Categories (15, object): [' ?', ' Adm-clerical', ' Armed-Forces', ' Craft-repair', ..., ' Protective-serv', ' Sales', ' Tech-support', ' Transport-moving']

In [99]:
lista_ocupaciones=salarios["occupation"].unique()
for valor in lista_ocupaciones:
    print(valor)

 Adm-clerical
 Exec-managerial
 Handlers-cleaners
 Prof-specialty
 Other-service
 Sales
 Craft-repair
 Transport-moving
 Farming-fishing
 Machine-op-inspct
 Tech-support
 ?
 Protective-serv
 Armed-Forces
 Priv-house-serv


In [100]:
len(lista_ocupaciones)

15

In [101]:
lista_ocupaciones_corregidas=["Adm-clerical", "Exec-managerial", "Handlers-cleaners", "Prof-specialty",
                              "Other-service", "Sales", "Craft-repair", "Transport-moving", "Farming-fishing",
                              "Machine-op-inspct", "Tech-support", "Desconocido", "Protective-serv",
                              "Armed-Forces","Priv-house-serv"]

In [102]:
for i in range(0,15):
    salarios=salarios.replace({lista_ocupaciones[i]:lista_ocupaciones_corregidas[i]})

/var/folders/qt/7mkpfxw95y54xs4g2ztqnb_c0000gn/T/ipykernel_62565/2699953996.py:2: FutureWarning: The behavior of Series.replace (and DataFrame.replace) with CategoricalDtype is deprecated. In a future version, replace will only be used for cases that preserve the categories. To change the categories, use ser.cat.rename_categories instead.
  salarios=salarios.replace({lista_ocupaciones[i]:lista_ocupaciones_corregidas[i]})


In [103]:
lista_ocupaciones=salarios["occupation"].unique()
for valor in lista_ocupaciones:
    print(valor)

Adm-clerical
Exec-managerial
Handlers-cleaners
Prof-specialty
Other-service
Sales
Craft-repair
Transport-moving
Farming-fishing
Machine-op-inspct
Tech-support
Desconocido
Protective-serv
Armed-Forces
Priv-house-serv


In [104]:
salarios['relationship'] = salarios['relationship'].str.strip()

In [105]:
frase="   hola a   todos"
frase.strip()

'hola a   todos'

In [106]:
lista_relationships=salarios["relationship"].unique()
for valor in lista_relationships:
    print(valor)

Not-in-family
Husband
Wife
Own-child
Unmarried
Other-relative


In [107]:
#Ordenar de acuerdo con la variable hours-per-week de manera ascendente
salarios.sort_values("hours-per-week", ascending=True)

age          workclass  fnlwgt      education  education-num  \
19750   23            Private   72887        HS-grad              9   
25078   74            Private  260669           10th              6   
11451   27            Private  147951        HS-grad              9   
8447    67        Desconocido  244122      Assoc-voc             11   
32525   81        Desconocido  120478      Assoc-voc             11   
...    ...                ...     ...            ...            ...   
25806   49        Desconocido   31478        7th-8th              4   
30610   51   Self-emp-not-inc   24790   Some-college             10   
10143   35        Desconocido  164866           10th              6   
4308    35   Self-emp-not-inc  166416        HS-grad              9   
1887    55   Self-emp-not-inc  184425   Some-college             10   

            marital-status         occupation    relationship  \
19750        Never-married       Craft-repair       Own-child   
25078             Divorced      Other-service   Not-in-family   
11451        Never-married  Machine-op-inspct  Other-relative   
8447               Widowed        Desconocido   Not-in-family   
32525             Divorced        Desconocido       Unmarried   
...                    ...                ...             ...   
25806   Married-civ-spouse        Desconocido         Husband   
30610   Married-civ-spouse    Exec-managerial         Husband   
10143             Divorced        Desconocido   Not-in-family   
4308    Married-civ-spouse    Farming-fishing         Husband   
1887    Married-civ-spouse    Farming-fishing         Husband   

                      race      sex  capital-gain  capital-loss  \
19750   Asian-Pac-Islander     Male             0             0   
25078                White   Female             0             0   
11451                White     Male             0             0   
8447                 White   Female             0             0   
32525                White   Female             0             0   
...                    ...      ...           ...           ...   
25806                White     Male             0             0   
30610                White     Male             0             0   
10143                White     Male             0             0   
4308                 White     Male             0             0   
1887                 White     Male             0             0   

       hours-per-week  native-country  salary  hours-per-day  
19750               1         Vietnam   <=50K            0.2  
25078               1   United-States   <=50K            0.2  
11451               1   United-States   <=50K            0.2  
8447                1   United-States   <=50K            0.2  
32525               1     Desconocido   <=50K            0.2  
...               ...             ...     ...            ...  
25806              99   United-States   <=50K           19.8  
30610              99   United-States    >50K           19.8  
10143              99   United-States   <=50K           19.8  
4308               99   United-States   <=50K           19.8  
1887               99   United-States    >50K           19.8  

[32561 rows x 16 columns]

In [108]:
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status         occupation   relationship    race      sex  \
0        Never-married       Adm-clerical  Not-in-family   White     Male   
1   Married-civ-spouse    Exec-managerial        Husband   White     Male   
2             Divorced  Handlers-cleaners  Not-in-family   White     Male   
3   Married-civ-spouse  Handlers-cleaners        Husband   Black     Male   
4   Married-civ-spouse     Prof-specialty           Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  \
0          2174             0              40   United-States   <=50K   
1             0             0              13   United-States   <=50K   
2             0             0              40   United-States   <=50K   
3             0             0              40   United-States   <=50K   
4             0             0              40            Cuba   <=50K   

   hours-per-day  
0            8.0  
1            2.6  
2            8.0  
3            8.0  
4            8.0

In [109]:
salarios.sort_values("native-country", ascending=True)

age          workclass  fnlwgt      education  education-num  \
21541   27            Private  169958        5th-6th              3   
18277   90            Private  311184      Bachelors             13   
27305   34   Self-emp-not-inc  114185      Bachelors             13   
5854    40            Private   37618   Some-college             10   
27299   37   Self-emp-not-inc  327164   Some-college             10   
...    ...                ...     ...            ...            ...   
23814   41   Self-emp-not-inc  145441      Assoc-voc             11   
7287    35            Private  164526        HS-grad              9   
31519   29            Private  273051      Bachelors             13   
4447    25            Private  191230   Some-college             10   
18247   41            Private  324629      Bachelors             13   

            marital-status        occupation   relationship    race      sex  \
21541        Never-married      Craft-repair      Own-child   White     Male   
18277   Married-civ-spouse             Sales        Husband   White     Male   
27305             Divorced  Transport-moving  Not-in-family   White     Male   
5854    Married-civ-spouse      Craft-repair        Husband   White     Male   
27299   Married-civ-spouse      Craft-repair        Husband   White     Male   
...                    ...               ...            ...     ...      ...   
23814   Married-civ-spouse      Craft-repair        Husband   White     Male   
7287    Married-civ-spouse     Other-service        Husband   White     Male   
31519   Married-civ-spouse   Exec-managerial        Husband   White     Male   
4447         Never-married   Exec-managerial      Own-child   White   Female   
18247   Married-civ-spouse   Exec-managerial        Husband   White     Male   

       capital-gain  capital-loss  hours-per-week native-country  salary  \
21541             0             0              40    Desconocido   <=50K   
18277             0             0              20    Desconocido   <=50K   
27305             0             0              50    Desconocido   <=50K   
5854              0             0              40    Desconocido   <=50K   
27299             0             0              70    Desconocido   <=50K   
...             ...           ...             ...            ...     ...   
23814             0             0              40     Yugoslavia   <=50K   
7287              0             0              40     Yugoslavia    >50K   
31519             0             0              52     Yugoslavia    >50K   
4447              0             0              40     Yugoslavia   <=50K   
18247             0             0              40     Yugoslavia   <=50K   

       hours-per-day  
21541            8.0  
18277            4.0  
27305           10.0  
5854             8.0  
27299           14.0  
...              ...  
23814            8.0  
7287             8.0  
31519           10.4  
4447             8.0  
18247            8.0  

[32561 rows x 16 columns]

In [110]:
salarios.keys()

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'salary', 'hours-per-day'],
      dtype='object')

In [111]:
salarios[salarios["sex"]==" Male"]["marital-status"]

0              Never-married
1         Married-civ-spouse
2                   Divorced
3         Married-civ-spouse
7         Married-civ-spouse
                ...         
32553          Never-married
32554     Married-civ-spouse
32555          Never-married
32557     Married-civ-spouse
32559          Never-married
Name: marital-status, Length: 21790, dtype: category
Categories (7, object): [' Divorced', ' Married-AF-spouse', ' Married-civ-spouse', ' Married-spouse-absent', ' Never-married', ' Separated', ' Widowed']

In [112]:
salarios["sex"].unique()

[' Male', ' Female']
Categories (2, object): [' Female', ' Male']

In [113]:
salarios["sex"]=salarios["sex"].replace({" Male": "Male", " Female": "Female"})

/var/folders/qt/7mkpfxw95y54xs4g2ztqnb_c0000gn/T/ipykernel_62565/2192342957.py:1: FutureWarning: The behavior of Series.replace (and DataFrame.replace) with CategoricalDtype is deprecated. In a future version, replace will only be used for cases that preserve the categories. To change the categories, use ser.cat.rename_categories instead.
  salarios["sex"]=salarios["sex"].replace({" Male": "Male", " Female": "Female"})


In [114]:
salarios[salarios["sex"]=="Male"]

age          workclass  fnlwgt      education  education-num  \
0       39          State-gov   77516      Bachelors             13   
1       50   Self-emp-not-inc   83311      Bachelors             13   
2       38            Private  215646        HS-grad              9   
3       53            Private  234721           11th              7   
7       52   Self-emp-not-inc  209642        HS-grad              9   
...    ...                ...     ...            ...            ...   
32553   32            Private  116138        Masters             14   
32554   53            Private  321865        Masters             14   
32555   22            Private  310152   Some-college             10   
32557   40            Private  154374        HS-grad              9   
32559   22            Private  201490        HS-grad              9   

            marital-status         occupation   relationship  \
0            Never-married       Adm-clerical  Not-in-family   
1       Married-civ-spouse    Exec-managerial        Husband   
2                 Divorced  Handlers-cleaners  Not-in-family   
3       Married-civ-spouse  Handlers-cleaners        Husband   
7       Married-civ-spouse    Exec-managerial        Husband   
...                    ...                ...            ...   
32553        Never-married       Tech-support  Not-in-family   
32554   Married-civ-spouse    Exec-managerial        Husband   
32555        Never-married    Protective-serv  Not-in-family   
32557   Married-civ-spouse  Machine-op-inspct        Husband   
32559        Never-married       Adm-clerical      Own-child   

                      race   sex  capital-gain  capital-loss  hours-per-week  \
0                    White  Male          2174             0              40   
1                    White  Male             0             0              13   
2                    White  Male             0             0              40   
3                    Black  Male             0             0              40   
7                    White  Male             0             0              45   
...                    ...   ...           ...           ...             ...   
32553   Asian-Pac-Islander  Male             0             0              11   
32554                White  Male             0             0              40   
32555                White  Male             0             0              40   
32557                White  Male             0             0              40   
32559                White  Male             0             0              20   

       native-country  salary  hours-per-day  
0       United-States   <=50K            8.0  
1       United-States   <=50K            2.6  
2       United-States   <=50K            8.0  
3       United-States   <=50K            8.0  
7       United-States    >50K            9.0  
...               ...     ...            ...  
32553          Taiwan   <=50K            2.2  
32554   United-States    >50K            8.0  
32555   United-States   <=50K            8.0  
32557   United-States    >50K            8.0  
32559   United-States   <=50K            4.0  

[21790 rows x 16 columns]

In [115]:
salarios["salary"].unique()

[' <=50K', ' >50K']
Categories (2, object): [' <=50K', ' >50K']

Modificar la variable salary de modo que los valores "<=50K" sean 0s y ">50" sean 1s

In [116]:
salarios["salary"]=salarios["salary"].replace({" <=50K": 0, " >50K": 1})
salarios.head()

/var/folders/qt/7mkpfxw95y54xs4g2ztqnb_c0000gn/T/ipykernel_62565/641816486.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  salarios["salary"]=salarios["salary"].replace({" <=50K": 0, " >50K": 1})
/var/folders/qt/7mkpfxw95y54xs4g2ztqnb_c0000gn/T/ipykernel_62565/641816486.py:1: FutureWarning: The behavior of Series.replace (and DataFrame.replace) with CategoricalDtype is deprecated. In a future version, replace will only be used for cases that preserve the categories. To change the categories, use ser.cat.rename_categories instead.
  salarios["salary"]=salarios["salary"].replace({" <=50K": 0, " >50K": 1})


age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status         occupation   relationship    race     sex  \
0        Never-married       Adm-clerical  Not-in-family   White    Male   
1   Married-civ-spouse    Exec-managerial        Husband   White    Male   
2             Divorced  Handlers-cleaners  Not-in-family   White    Male   
3   Married-civ-spouse  Handlers-cleaners        Husband   Black    Male   
4   Married-civ-spouse     Prof-specialty           Wife   Black  Female   

   capital-gain  capital-loss  hours-per-week  native-country salary  \
0          2174             0              40   United-States      0   
1             0             0              13   United-States      0   
2             0             0              40   United-States      0   
3             0             0              40   United-States      0   
4             0             0              40            Cuba      0   

   hours-per-day  
0            8.0  
1            2.6  
2            8.0  
3            8.0  
4            8.0


1.   ¿Cuál es la edad promedio de los empleados?
2.   ¿En qué porcentajes se distribuyen las diferentes raza?
3. De las personas que ganan más de 50 mil, ¿cuántas son mujeres?
4. Analizar la relación entre el nivel de estudios y los sueldos




In [123]:
promedio_edades=salarios["age"].mean()
print("La edad promedio de los empleados es de aproximadamente", round(promedio_edades), "años")

La edad promedio de los empleados es de aproximadamente 39 años


In [124]:
salarios["age"].count()

np.int64(32561)

In [125]:
salarios["race"].unique()

[' White', ' Black', ' Asian-Pac-Islander', ' Amer-Indian-Eskimo', ' Other']
Categories (5, object): [' Amer-Indian-Eskimo', ' Asian-Pac-Islander', ' Black', ' Other', ' White']

In [126]:
porcentajes_raza = salarios["race"].value_counts(normalize=True) * 100
print(porcentajes_raza)

race
White                 85.427352
Black                  9.594300
Asian-Pac-Islander     3.190934
Amer-Indian-Eskimo     0.955130
Other                  0.832284
Name: proportion, dtype: float64


In [122]:
tabla_estudios_ingresos = pd.crosstab(salarios["education"], salarios["salary"], normalize="index") * 100
print(tabla_estudios_ingresos)

salary                 0          1
education                          
10th           93.354770   6.645230
11th           94.893617   5.106383
12th           92.378753   7.621247
1st-4th        96.428571   3.571429
5th-6th        95.195195   4.804805
7th-8th        93.808050   6.191950
9th            94.747082   5.252918
Assoc-acdm     75.164011  24.835989
Assoc-voc      73.878437  26.121563
Bachelors      58.524743  41.475257
Doctorate      25.907990  74.092010
HS-grad        84.049138  15.950862
Masters        44.341265  55.658735
Preschool     100.000000   0.000000
Prof-school    26.562500  73.437500
Some-college   80.976546  19.023454
